### PR TITLE
Updated vmware_host_facts and rewrite of vmware_dns_config

### PR DESCRIPTION
**Pull request by ryancbutler**
##### SUMMARY
- updated **vmware_host_facts**
  - Added maintenance mode and 
  - Added uptime

- Rewrite **vmware_dns_config**
  - Rewrote to allow cluster parameter
  - Added search domain
  - Added DHCP option

I'm not sure how the author field should be handled. Let me know if any adjustments need to be made.

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_host_facts.py [hunk 1]
```diff
@@ -277,6 +279,8 @@ def get_system_facts(self):
             'ansible_product_serial': sn,
             'ansible_bios_date': self.host.hardware.biosInfo.releaseDate,
             'ansible_bios_version': self.host.hardware.biosInfo.biosVersion,
+            'ansible_uptime': self.host.summary.quickStats.uptime,
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_host_facts.py:282]: Could you please create a separate PR for this change ? Since these two changes are not related  to each other.

File: lib/ansible/modules/cloud/vmware/vmware_dns_config.py [hunk 2]
```diff
@@ -20,106 +20,300 @@
 version_added: 2.0
 author:
 - Joseph Callen (@jcpowermac)
+- Ryan Butler (@ryan_c_butler)
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_dns_config.py:23]: Would it be possible for you to work on https://github.com/ansible/ansible/pull/47271 which tries to deprecated `vmware_dns_config` and create a new module called `vmware_host_dns` ? Let me know ? Thanks.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_dns_config.py (hunk 2 above)
Comment (ryancbutler) [reply lib/ansible/modules/cloud/vmware/vmware_dns_config.py:23]: I guess I should have looked at the open PRs :) Let me know how I can help.

File: lib/ansible/modules/cloud/vmware/vmware_host_facts.py (hunk 1 above)
Comment (ryancbutler) [reply lib/ansible/modules/cloud/vmware/vmware_host_facts.py:282]: Sounds good

**Comment by ryancbutler** [comment]
Closing

File: lib/ansible/modules/cloud/vmware/vmware_dns_config.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_dns_config.py:23]: Thanks, you can create a new PR using the code available in #47271, test it and let me know.
